docs/guides/admin/docs/configuration/security.jwt.md

```diff
@@ -1,18 +1,107 @@
-[JSON Web Tokens (JWTs)](https://datatracker.ietf.org/doc/html/rfc7519). With this feature, a login-mechanism based
-on the [OpenID Connect (OIDC)](https://openid.net/connect/) protocol can be configured, since OIDC uses JWTs.
+This page describes how to configure Opencast to enable authentication and/or authorization based on
+[JSON Web Tokens (JWTs)](https://datatracker.ietf.org/doc/html/rfc7519). JWTs can be used in various ways:
+
+- With static file authorization (particularly useful with external tools).
+- External apps can send users to Opencast (e.g. for Studio or Editor) with a session that has certain roles.
+- A login-mechanism based on the [OpenID Connect (OIDC)](https://openid.net/connect/) protocol can be configured.
+
+*Note*: Some of these use cases might not be perfectly supported by Opencast yet, but are possible in principle.
 
 Prerequisites
 -------------
 
-This guide assumes that a JWT provider is already setup and Opencast receives the JWT within an HTTP request header,
-or in a request parameter (i.e. a query parameter for `GET`-requests, and a form parameter for `POST`-requests).
+Opencast does not generate JWTs, but reads them in incoming requests. This guide assumes that you already have a JWT provider, i.e. another services that generates JWTs for consumption in Opencast. This could be [Tobira](https://elan-ev.github.io/tobira/setup/auth/jwt), an LMS, an OIDC provider, or something else. These generally have their own setup guides. The JWT has to be passed to Opencast via HTTP header or via request parameter (i.e. a query parameter for `GET`-requests, and a form parameter for `POST`-requests).
+
 In order to integrate Opencast with an OIDC provider you could use the
 [oauth2-proxy](https://github.com/oauth2-proxy/oauth2-proxy).
 
+Standard OC Schema for JWTs
+---------------------------
+
+After the signature verification, a JWT is just a bunch of "claims" (JSON fields). How to interpret these claims is completely up to the communicating parties. Protocols like OIDC define exactly which claims are to be used and how. For other cases where external Opencast apps (like Tobira or an LMS) need to communicate with Opencast, this section defines a standard schema to follow. You can configure Opencast differently, but this aims at standardizing JWT usage across Opencast-related applications and offer an out-of-the-box solution.
+
+This schema is designed to be flexible and support all current and conceivable future use cases, while keeping the JWT size for each use case small.
+
+### Quick overview
+
+```json5
+{
+    // Standard claims
+    "exp": 1499863217, // required
+    "nbf": 1548632170, // optional
+
+    // User info
+    "username": "jose",
+    "name": "José Carreño Quiñones",
+    "email": "jose@example.com",
+
+    // Authorization, i.e. privileges the request/user should have
+    "roles": ["STUDIO", "API_EVENTS_VIEW"], // without `ROLE_` prefix
```

Comment:
I know that you want to keep the JWT short, but I would like to avoid magic steps. As such, I'm in favor of always including full role names.

Reply:
Ok lets use this thread to continue the discussion from DACH: people felt uncomfortable omitting the prefix. Lars even seemed to say that we can't? Can someone explain? 

Because I approached this from the other side multiple sides: I asked whether I could use roles in ACLs that didn't start with `ROLE_` (e.g. for the magic roles in [this PR](https://github.com/opencast/opencast/pull/5056)). And there, I got the answer "hell no, please always use the `ROLE_` prefix!". So what is it now? Either we are prohibiting us from using anything not starting with `ROLE_`, but then we should be able to rely on this in scenarios like this JWT schema. Or we cannot rely on it, but then we should be able to use roles without that prefix in scenarios where it makes sense. Restricting ourselves in both situations makes no sense to me. 
Also note that this is just about JWTs, so worst case is that you cannot grant roles not starting with `ROLE_` via JWT. It doesn't affect any other part of Opencast.

Reply:
What exactly is the argument for removing `ROLE_`? I only see shorter tokens, which I hope is not really a problem.

What about reducing "magic" steps?

Regarding the prefix: [You can configure the prefix](https://github.com/opencast/opencast/blob/develop/etc/org.opencastproject.userdirectory.UserIdRoleProvider.cfg#L6-L8) used for user roles, i.e. `ROLE_USER_`. Not sure if people do that and also not sure if they then don't use `ROLE_`, but you could. I don't think you can configure other role prefixes (for groups, normal roles, etc.), but you can set roles without `ROLE_` for events. Roles typically set by LTI have the form `{number}_(LEARNER|INSTRUCUTOR)` with no prefix (`number` is then usually the LMS course ID).

Reply:
We also have roles without `ROLE_` Prefix like:
- `LDAP_XYZ_ADMIN` which are derived from a ldap attribute, which can be configured here: https://github.com/opencast/opencast/blob/develop/etc/org.opencastproject.userdirectory.ldap.cfg.template#L85-L89
- `XYZ_PUBLIC_VIEW` which we also use to determine if a video is public

I agree with @mtneug on this. "Magic" could lead to unexpected issues here.

Reply:
Alright, i was not aware of people using roles without that prefix. It seems like there is plenty of precedence. As to my previous point: then we should have not restricted us previously to only use roles starting with `ROLE_`. But oh well, that happened now. 

Will remove the "prefix is added" part of the schema definition soon.